aws/cloudformation-template-ipsec

```diff
@@ -0,0 +1,821 @@
+                                "       keyMaterial = response['KeyMaterial']",
+                                "       cfnresponse.send(event, context, cfnresponse.SUCCESS, {'KeyMaterial':keyMaterial, 'KeyName':keyName}, 'KeyPairInfo')",
+                                "   except Exception:",
+                                "       cfnresponse.send(event, context, cfnresponse.FAILED, {})"
+                            ]
+                        ]
+                    }
+                }
+            },
+            "Metadata": {
+                "AWS::CloudFormation::Designer": {
+                    "id": "99fce86e-18b8-4b1b-a572-7bef3c5cece7"
+                }
+            },
+            "DependsOn": [
+                "LambdaExecutionRole"
+            ]
+        },
+        "AMIInfo": {
+            "Type": "Custom::AMIInfo",
+            "Properties": {
+                "Region": {
+                    "Ref": "AWS::Region"
+                },
+                "ServiceToken": {
+                    "Fn::GetAtt": [
+                        "AMIInfoFunction",
+                        "Arn"
+                    ]
+                },
+                "Distribution": {
+                    "Ref": "OS"
+                }
+            },
+            "Metadata": {
+                "AWS::CloudFormation::Designer": {
+                    "id": "2c5cc5a9-5a17-4d54-80ea-56e204c9c1a1"
+                }
+            },
+            "DependsOn": [
+                "AMIInfoFunction"
+            ]
+        },
+        "AMIInfoFunction": {
+            "Type": "AWS::Lambda::Function",
+            "Properties": {
+                "Handler": "index.handler",
+                "Runtime": "python3.7",
+                "Role": {
+                    "Fn::GetAtt": [
+                        "LambdaExecutionRole",
+                        "Arn"
+                    ]
+                },
+                "Code": {
+                    "ZipFile": {
+                        "Fn::Join": [
+                            "\n",
+                            [
+                                "import boto3",
+                                "import cfnresponse",
+                                "def handler(event, context):",
+                                "   try:",
+                                "       regionName = event['ResourceProperties']['Region']",
+                                "       distribution = event['ResourceProperties']['Distribution']",
+                                "       ec2 = boto3.client('ec2',regionName)",
+                                "       IAMName = ''",
+                                "       if distribution == 'Ubuntu16.04':",
+                                "           IAMName = 'ami-ubuntu-16.04-1.12.0-00-1538138520'",
+                                "       elif distribution == 'Ubuntu18.04':",
+                                "           IAMName = 'ami-ubuntu-18.04-1.16.2-00-1571834140'",
```

Comment:
Are these official Amazon Machine Images (AMIs) from Canonical? I can't find them here:
https://cloud-images.ubuntu.com/locator/

Please make sure to only use official AMIs, preferably latest versions, in this template.

Reply:
Thanks for point it out. I will replace it with a script that sorts out the official images in respect to their release dates.